intelmq/bots/parsers/alienvault/parser_otx.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+"""
+; Events are gathered based on user subscriptions in AlienVault OTX
+; The data structure is described in detail here:
+; https://github.com/AlienVault-Labs/OTX-Python-SDK/blob/master/howto_use_python_otx_api.ipynb
+"""
+from __future__ import unicode_literals
+import sys
+import json
+
+from intelmq.lib import utils
+from intelmq.lib.bot import Bot
+from intelmq.lib.harmonization import DateTime
+from intelmq.lib.message import Event
+
+HASHES = {
+    'FileHash-SHA256': '$5$',
+    'FileHash-SHA1': '$sha1$',
+    'FileHash-MD5': '$1$'
+}
+
+class AlienVaultOTXParserBot(Bot):
+
+    def process(self):
+        report = self.receive_message()
+        if (report is None or not report.contains("raw")):
+            self.acknowledge_message()
+            return
+
+        raw_report = utils.base64_decode(report.value("raw"))
+
+        for pulse in json.loads(raw_report):
+            additional = json.dumps(
+                {"author": pulse['author_name'], "pulse": pulse['name']})
+            for indicator in pulse["indicators"]:
+                event = Event()
+                # hashes
+                if indicator["type"] in HASHES.keys():
+                    event.add('malware.hash',
+                              HASHES[indicator["type"]] + indicator["indicator"])
+                # fqdn
+                if indicator["type"] in ['hostname', 'domain']:
+                    event.add('source.fqdn',
+                              indicator["indicator"], sanitize=True)
+                # IP addresses
+                elif indicator["type"] in ['IPv4', 'IPv6']:
+                    event.add('source.ip',
+                              indicator["indicator"], sanitize=True)
+                # emails
+                elif indicator["type"] == 'email':
+                    event.add('source.account',
+                              indicator["indicator"], sanitize=True)
+                # URLs
+                elif indicator["type"] in ['URL', 'URI']:
+                    event.add('source.url',
+                              indicator["indicator"], sanitize=True)
+                # CIDR
+                elif indicator["type"] in ['CIDR']:
+                    event.add('source.network',
+                              indicator["indicator"], sanitize=True)
+                # FilePath, Mutex, CVE - TODO: process these IoCs as well
+                else:
+                    continue
+
+                event.add('comment', pulse['description'])
+                event.add('additional', additional, sanitize=True)
+                event.add('classification.type', 'blacklist', sanitize=True)
+                event.add('time.observation', report.value(
```

Comment:
@SYNchroACK there seem to be a typo in that example```reports``` instead of report ```report```

@sebix The proposal of yours also takes the "raw" field from the report making it impossible to add the "raw" record for the event itself later, failing later like this:
```ERROR - Bot has found a problem.\nTraceback (most recent call last):\n  File "/usr/local/lib/python2.7/dist-packages/intelmq-1.0.0-py2.7.egg/intelmq/lib/bot.py", line 117, in start\n    self.process()\n  File "/usr/local/lib/python2.7/dist-packages/intelmq-1.0.0-py2.7.egg/intelmq/bots/parsers/alienvault/parser_otx.py", line 75, in process\n    event.add("raw", json.dumps(indicator), sanitize=True)\n  File "/usr/local/lib/python2.7/dist-packages/intelmq-1.0.0-py2.7.egg/intelmq/lib/message.py", line 77, in add\n    raise exceptions.KeyExists(key)\nKeyExists: key u\'raw\' already exists```

Should we replace the raw field instead of adding (does not feel like a proper way copying the whole raw report again and again) it or change the "constructor" to ignore the "raw" attribute from report (sorry for the cpp terminology).